django/db/models/query.py

```diff
@@ -1137,6 +1130,21 @@ def _has_filters(self):
         """
         return self.query.has_filters()
 
+    def _check_allowed(self, action):
+        require_no_limits = {
```

Comment:
A name like `action_to_msg`makes more sense to me. Is it worth putting it at the module level to avoid constructing that dict each time the method is called?